projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php

```diff
@@ -355,6 +356,37 @@ function wpcom_get_domain_upsell_badge_text() {
 	return wpcom_is_checklist_task_complete( 'domain_upsell' ) ? '' : __( 'Upgrade plan', 'jetpack-mu-wpcom' );
 }
 
+/**
+ * Determines whether or not domain upsell task should be visible.
+ *
+ * @return bool True if user is on a free plan with no custom domain name.
+ */
+function wpcom_is_domain_upsell_visible() {
+	if ( ! class_exists( '\A8C\Billingdaddy\Container' ) ) {
+		return false;
+	}
```

Comment:
The visibility condition that I was looking for is: `User is on a free plan with no custom domain name.`, I've added this check since I don't think it's possible to be on a free plan on a non-simple site. 

Maybe I am missing something?